frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_item.scss

```diff
@@ -19,68 +19,62 @@
 }
 
 .flex-element-menu {
-  @include flexColumn;
-  gap: $s-12;
-  margin-block-end: $s-8;
+  display: grid;
+  grid-template-columns: repeat(8, var(--sp-xxxl));
+  gap: var(--sp-xs);
 }
 
 .behaviour-menu {
-  display: flex;
-  gap: $s-4;
-}
-
-.horizontal-behaviour {
-  &.one-element {
-    width: $s-28;
-  }
-  &.two-element {
-    width: $s-60;
-  }
-  &.three-element {
-    width: $s-92;
-  }
+  display: grid;
+  column-gap: var(--sp-xs);
+  grid-template-columns: auto auto;
 }
 
 .vertical-behaviour {
   .rotated {
     transform: rotate(90deg);
   }
-  &.one-element {
-    width: $s-28;
-  }
-  &.two-element {
-    width: $s-60;
-  }
-  &.three-element {
-    width: $s-92;
-  }
 }
 
 .z-index-wrapper {
   @extend .input-element;
   @include bodySmallTypography;
-  width: $s-60;
+  grid-column: 6 / span 3;
 }
 
 .row {
+  border: 1px solid red;
   display: flex;
-  gap: $s-4;
+  gap: var(--sp-xs);
+}
+
+.position-row,
+.behavior-row,
+.align-row,
+.margin-row {
+  display: grid;
+  grid-template-columns: subgrid;
+  grid-column: 1 / -1;
 }
 
 .position-options {
-  width: $s-188;
+  width: 100%;
+  grid-column: 1 / span 5;
 }
 
 .margin-row {
-  display: flex;
-  align-items: flex-start;
-  gap: $s-4;
+  display: grid;
+  // NOTE: this does not follow the 8-column grid and the size is not normalized,
+  --input-width: calc(var(--sp-xxxl) * 3.5 + 3 * var(--sp-xs) - var(--sp-xs) / 2);
+  grid-template-columns: var(--input-width) var(--input-width) var(--sp-xxxl);
+  gap: var(--sp-xs);
 }
 
 .margin-mode {
   @extend .button-tertiary;
+  grid-column: 3;
   height: $s-32;
-  width: $s-28;
+  width: $s-32;
```

Comment:
We shouldn't need this width `grid-column: 3 span 1` if `grid-column: 3` is not enough